builtin/providers/vcd/resource_vcd_firewall_rules.go

```diff
@@ -0,0 +1,241 @@
+package vcd
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hmrc/vmware-govcd"
+	types "github.com/hmrc/vmware-govcd/types/v56"
+	"strings"
+)
+
+func resourceVcdFirewallRules() *schema.Resource {
```

Comment:
Hi, the reason for grouping these was for situations when you need rules in a certain order. You could have multiple resources with one or more rules. With vCloud Director all firewall rules are applied on the edge gateway, previously tools have dealt with this in a destructive way, each set of rules overwriting the last. We have tried to get around this by checking the existing set and applying on top. I hope the changes to the documentation example will show this better.